### DISCUSSION TITLE
Identical request not working with ky but working with fetch

### DISCUSSION THREAD

**Discussion by schneedorfer** [sindresorhus/ky#520] [Q&A]
Hi, I have defined a route in Next.js 13 where I fetch external API. However an identical request fails with ky but works with fetch, what could I be doing wrong here? 

I am using the app router and `ky-universal`. The error message from the external API says that the email and password is undefined in the request body. However, when I check the request in the catch block, it does have both fields set correctly.

**The request using ky:**
``` ts
const res = await ky.post("auth/native", {
  json: credentials,
  prefixUrl: `${env("baseUrl")}/api/rest/v1`,
  headers: {
    APIKey: env("apiKey") ?? "",
  },
});
```

**The request using fetch:**
``` ts
const res = await fetch(`${env("baseUrl")}/api/rest/v1/auth/native`, {
  body: JSON.stringify(credentials),
  method: "POST",
  headers: {
    "Content-Type": "application/json",
    APIKey: env("apiKey") ?? "",
  },
});
```

**Comment by sindresorhus** [comment]
I would use an app like [Proxyman](https://proxyman.io) to inspect what the actual request is for both.

**Reply by schneedorfer** [reply]
I use Ubuntu, but I guess I will try to find an alternative. Thanks for the tip.

**Comment by schneedorfer** [comment]
So I am here with an update. The request works if I pass the fetch function that Next.js polyfills.
``` ts
const res = await ky.post("auth/native", {
  fetch, // added line that fixes the issue
  json: credentials,
  prefixUrl: `${env("baseUrl")}/api/rest/v1`,
  headers: {
    APIKey: env("apiKey") ?? "",
  },
});
```

Unfortunately, I cannot add this to the whole API client that I am creating since this throws an error when used in browser environment.
`Failed to execute 'fetch' on 'Window': Illegal invocation`

So I had to create a special instance for the server like so:
``` ts
export const publicApi = ky.extend({
  prefixUrl: `${env("baseUrl")}/api/rest/v1`,
  hooks: {
    beforeRequest: [attachApiKeyHook],
  },
});

export const ssrPublicApi = publicApi.extend({ fetch });
```

This is quite annoying since I would have to keep double the amount of api instances, If anyone knows a better solution please share it with me.

**Reply by sholladay** [reply]
So the problem is that Ky is not using the appropriate fetch function? Which fetch is it using and do you know why?
